source/vistas/core/graphics/feature/factory.py

```diff
@@ -0,0 +1,234 @@
+import os
+from functools import partial
+
+import numpy
+import pyproj
+import shapely.geometry as shp
+from shapely.ops import transform
+from triangle import triangulate
+
+from vistas.core.color import RGBColor
+from vistas.core.gis.elevation import ElevationService, TILE_SIZE, meters_per_px
+from vistas.core.graphics.factory import MapMeshFactory, MeshFactoryWorker
+from vistas.core.graphics.feature.geometry import FeatureGeometry
+from vistas.core.graphics.feature.shader import FeatureShaderProgram
+from vistas.core.graphics.mesh import Mesh
+from vistas.core.plugins.data import FeatureDataPlugin
+
+
+class FeatureFactoryWorker(MeshFactoryWorker):
+
+    task_name = "Building Features"
+
+    def work(self):
+        verts, indices, normals, colors = [None] * 4
+        if self.factory.needs_vertices and not self.task.should_stop:
+            verts, indices, normals = self.factory.generate_meshes(self.task)
+            self.factory.needs_vertices = False
+
+        if self.factory.needs_color and not self.task.should_stop:
+            colors = self.factory.generate_colors(self.task)
+            self.factory.needs_color = False
+
+        self.sync_with_main(
+            self.factory.update_features, kwargs=dict(vertices=verts, indices=indices, normals=normals, colors=colors),
+            block=True
+        )
+
+
+class FeatureFactory(MapMeshFactory):
+    """ A MapMeshFactory for handling polygon rendering. """
+
+    worker_class = FeatureFactoryWorker
+
+    def __init__(self, extent, data_src: FeatureDataPlugin, shader=None, plugin=None, initial_zoom=10):
+        if shader is None:
```

Comment:
Sweet! I didn't know you could do that.

Reply:
There will be a slight different in behavior though. In my example, it would choose `FeatureShaderProgram()` any time `shader` evaluates to `False`, not just when it's explicitly `None`. Subtle, but important.